app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperDownloader.kt

```diff
@@ -80,7 +80,7 @@ class WallpaperDownloader(
             if (!response.isSuccess) {
                 throw IllegalStateException()
             }
-            File(localFile.path.substringBeforeLast("/")).mkdirs()
+            localFile.parentFile?.mkdirs()
```

Comment:
This wouldn't help alone because the wallpaper code still has strong expectations that it's a unix system (see line 72).

Reply:
It's enough to output correct results for the current test suite. The path construction in line 72/74 appears to be correct even on Windows, and changing it to use backslashes actually causes test failures.

Reply:
The path on line 72 is a remote path for an HTTP request, so it is not platform dependent.

Reply:
At a glance though, I don't think this will work if the parent directory doesn't already exist, as `parentFile` will return null. Could you add a test that verifies that case?

The previous implementation worked by the following:
- create a `File` object with an explicit path
- create that `File` as a directory, including its parent directories

Reply:
`parentFile` only returns null when the `pathname` (i.e. the string) does not include a path separator, which should be impossible since `localFile` is defined as child file. To illustrate this, try out this passing test in a new file:

```kotlin
package org.mozilla.fenix.wallpapers

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import java.io.File

class WallTest {

    @Test
    fun test() {
        val file = File("testing/one/two/three")
        assertFalse("file does not exist", file.exists())
        assertFalse("file parent is still non-null since it just depends on the pathname", file.parentFile == null)
        assertFalse("file parent does not yet exist", file.parentFile!!.exists())
        file.parentFile?.mkdirs()
        assertTrue("file parent now exists", file.parentFile!!.exists())

    }
}
```

Reply:
Oh neat! Thanks for taking the time to walk me through that, I guess my understanding was wrong. This looks to be an overall improvement to me then, especially in terms of readability.